src/ext/java/org/opentripplanner/ext/siri/AddedTripBuilder.java

```diff
@@ -236,11 +244,26 @@ Result<TripUpdate, UpdateError> build() {
       return TripTimesValidationMapper.toResult(tripId, validityResult.get());
     }
 
+    var tripOnServiceDate = TripOnServiceDate
+      .of(tripId)
+      .withTrip(trip)
+      .withServiceDate(serviceDate)
+      .withReplacementFor(replacedTrips)
+      .build();
+
     // Adding trip to index necessary to include values in graphql-queries
     // TODO - SIRI: should more data be added to index?
     transitModel.getTransitModelIndex().getTripForId().put(tripId, trip);
     transitModel.getTransitModelIndex().getPatternForTrip().put(trip, pattern);
     transitModel.getTransitModelIndex().getPatternsForRoute().put(route, pattern);
+    transitModel
+      .getTransitModelIndex()
+      .getTripOnServiceDateById()
+      .put(tripOnServiceDate.getId(), tripOnServiceDate);
+    transitModel
+      .getTransitModelIndex()
+      .getTripOnServiceDateForTripAndDay()
+      .put(new TripIdAndServiceDate(tripId, serviceDate), tripOnServiceDate);
```

Comment:
Question: Is it threadsafe to modify these maps like we do here? It looks like these maps are being read by the apis, I suppose in different threads.

Reply:
it is not thread-safe and we get random errors in our production environment due to similar operations elsewhere in the code. This is a known issue.